pkg/apis/v1alpha1/dnspolicy_types.go

```diff
@@ -33,6 +33,36 @@ type DNSPolicySpec struct {
 
 	// +optional
 	HealthCheck *HealthCheckSpec `json:"healthCheck,omitempty"`
+
+	// +kubebuilder:validation:Required
+	// +required
+	LoadBalancing *LoadBalancingSpec `json:"loadBalancing"`
+}
+
+type LoadBalancingSpec struct {
+	// +required
+	Weighted *LoadBalancingWeighted `json:"weighted,omitempty"`
+	// +optional
+	Geo *LoadBalancingGeo `json:"geo,omitempty"`
+}
+
+type LoadBalancingWeighted struct {
+	// defaultWeight is the record weight to use when no other can be determined for a dns target cluster.
+	//
+	// +kubebuilder:default=120
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=120
```

Comment:
Max value supported by aws is 255 https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/resource-record-sets-values-weighted.html#rrsets-values-weighted-weight, google supports higher values up to 10000 but we probably want to keep this limited to something all providers will support, so this should probably be a max value of 255.

Reply:
Azure has  a max of 1000